### PR TITLE
vNext ready

### PR DESCRIPTION
**Pull request by danielmarbach**
This PR makes develop vNext ready. I splitted out almost all obsoletions into single commits in order to make it easier if we need to revert a specific commit. I'm going to make inline comments about a few things I detected.

@andreasohlund please review

### CODE REVIEW

**Comment by danielmarbach** [comment]
@Particular/engineers review please and merge when ready in order to make develop vNext ready

File: src/NServiceBus.Core/Encryption/WireEncryptedString.cs [hunk 1]
```diff
@@ -63,13 +51,13 @@ public EncryptedValue EncryptedValue
         /// <summary>
         /// Only kept for backwards compatibility reasons
         /// </summary>
-        [ObsoleteEx(TreatAsErrorFromVersion = "6.0",RemoveInVersion = "6.0",Message = "No longer required")]
+        [ObsoleteEx(TreatAsErrorFromVersion = "6.0", RemoveInVersion = "7.0", Message = "No longer required")]
```
Comment (SimonCropp) [inline src/NServiceBus.Core/Encryption/WireEncryptedString.cs]: Why is this not removed?

Comment (danielmarbach) [inline src/NServiceBus.Core/Encryption/WireEncryptedString.cs]: I thought the idea is for us to obsolete gradually by using TreatAsErrorFromVersion one version higher than RemoveInVersion. Is my assumption wrong?

File: src/NServiceBus.Core/Transports/Msmq/CorrelationIdMutatorForBackwardsCompatibilityWithV3.cs [hunk 2]
```diff
@@ -1,28 +0,0 @@
-﻿namespace NServiceBus.Transports.Msmq
-{
-    using System;
-    using MessageMutator;
-    using Unicast.Messages;
-
-    [ObsoleteEx(RemoveInVersion ="6.0")]
-    class CorrelationIdMutatorForBackwardsCompatibilityWithV3 : IMutateOutgoingTransportMessages
```
Comment (SimonCropp) [inline src/NServiceBus.Core/Transports/Msmq/CorrelationIdMutatorForBackwardsCompatibilityWithV3.cs]: @andreasohlund so how long do we want wirecompat with an older version?

File: src/NServiceBus.Core/Unicast/Messages/DeserializeLogicalMessagesBehavior.cs [hunk 3]
```diff
@@ -97,12 +92,6 @@ List<LogicalMessage> Extract(TransportMessage physicalMessage)
             }
         }
 
-        [ObsoleteEx(RemoveInVersion = "6.0")]
-        bool DoesTypeHaveImplAddedByVersion3(string existingTypeString)
```
Comment (SimonCropp) [inline src/NServiceBus.Core/Unicast/Messages/DeserializeLogicalMessagesBehavior.cs]: @andreasohlund same wirecompat question

File: src/NServiceBus.Core/Transports/Msmq/CorrelationIdMutatorForBackwardsCompatibilityWithV3.cs (hunk 2 above)
Comment (johnsimons) [inline src/NServiceBus.Core/Transports/Msmq/CorrelationIdMutatorForBackwardsCompatibilityWithV3.cs]: I say we re-evaluate in 1 years time, so bump it to v7.
By then hopefully not many customers are still in v3 land.

Comment (SimonCropp) [inline src/NServiceBus.Core/Transports/Msmq/CorrelationIdMutatorForBackwardsCompatibilityWithV3.cs]: @johnsimons sounds reasonable. r there any other changes in this u think should be handled in the same way?

Comment (johnsimons) [inline src/NServiceBus.Core/Transports/Msmq/CorrelationIdMutatorForBackwardsCompatibilityWithV3.cs]: Have not reviewed this PR yet, so not sure
But the things to watch out is breaking wire compat, that is all IMO.

File: src/NServiceBus.Core/Unicast/Messages/DeserializeLogicalMessagesBehavior.cs (hunk 3 above)
Comment (SimonCropp) [inline src/NServiceBus.Core/Unicast/Messages/DeserializeLogicalMessagesBehavior.cs]: @johnsimons assume this should be v7 as well?

Comment (johnsimons) [inline src/NServiceBus.Core/Unicast/Messages/DeserializeLogicalMessagesBehavior.cs]: Yep

Comment (danielmarbach) [inline src/NServiceBus.Core/Unicast/Messages/DeserializeLogicalMessagesBehavior.cs]: So basically I need to pull that in again. I'll do right after the talk with danny.

**Comment by danielmarbach** [comment]
There is a HostInformation acceptance test failing. I'll take a look later after that it should be good to go

**Comment by danielmarbach** [comment]
@johnsimons could you please have a look at this and merge it. Currently there is still a failing ATT `When_reply_from_a_finder.Should_be_received_by_handler`. Maybe you can take it over and merge when fixed. I'm not sure what this test has to do with the stuff I changed... Apart from that it is ready.

**Comment by danielmarbach** [comment]
Thanks!

On Wed, Feb 18, 2015 at 5:51 AM, John Simons <notifications@github.com>
wrote:

> Merged #2655 <https://github.com/Particular/NServiceBus/pull/2655>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/2655#event-237443829>.
>



-- 

Daniel Marbach

Solution Architect, Particular Software

Email:  daniel.marbach@particular.net

Web:    www.particular.net
<http://t.signauxneuf.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJN7t5XX4RzDjlVS9QB88pTc_dW3Ljg7x56dWyBdgBXs402?t=http%3A%2F%2Fwww.particular.net%2F&si=4896787407044608&pi=33cc6202-1f31-4040-8d6c-e70641f44bba>

**Pull request merged**